src/Protocol/ForgeMods.cpp

```diff
@@ -0,0 +1,89 @@
+
+// ForgeMods.cpp
+
+// Data structure listing the Forge mods name and versions
+
+#include "Globals.h"
+#include "ForgeMods.h"
+
+cForgeMods::cForgeMods()
+{
+}
+
+
+
+
+
+cForgeMods::cForgeMods(AStringMap a):m_Mods(a)
+{
+	for (const auto & item: a)
+	{
+		m_ModNames.push_back(item.first);
+	}
+}
```

Comment:
You're copying `a` twice here. You should either take it by const reference or you should move into `m_Mods` after filling `m_ModNames`.